docs/enhancements/api-for-skipping-validations.md

```diff
@@ -0,0 +1,153 @@
+{
+  "host-skipped-validations": [
+    "no-ip-collisions-in-network",
+    "sufficient-network-latency-requirement-for-role'"
+  ],
+  "cluster-skipped-validations": [
+    "network-prefix-valid",
+    "api-vips-defined"
+  ]
+}
+```
+
+### Add some skipped validations.
+
+POST api/v2/cluster/<id>/skipped-validations
+
+Essentially a document will be submitted via POST in JSON
+```
+{
+  "skipped-host-validations": [
+    "no-ip-collisions-in-network"
+  ],
+  "skipped-cluster-validations": [
+    "api-vips-defined"
+  ]
+}
+```
+
+* Any entries found will be added to the existing list of validations to be skipped. This will appropriately de-deuplicated and no error will be returned if duplication is attempted.
+* If an attempt is made to reference a validation that does not exist then a warning will be logged but this will not be considered fatal.
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+### Edit the validations
+
+PUT api/v2/cluster/<id>/skipped-validations
+
+Essentially a document will be submitted via POST in JSON
+```
+{
+  "skipped-host-validations": [
+    "no-ip-collisions-in-network"
+  ],
+  "skipped-cluster-validations": [
+    "api-vips-defined"
+  ]
+}
+```
+* The freshly submitted JSON document, will be the new source of truth. All existing "skip operations" will be undone before applying this.
+* If an attempt is made to reference a validation that does not exist then a warning will be logged but this will not be considered fatal.
+* Submitting an empty doc '{}' will lead to the deletion of all skipped validations.
+* After the call has completed, if this leads to a change in the collection of skipped validations then an event will be emitted.
+
+### Effect on validation.
+
+When a validation is skipped then this should be reflected in the Validation status. For example;
+```
+{
+    "id": "api-vips-defined",
+    "status": "skipped",
+    "message": "API virtual IPs validation has been suppressed by the user"
+}
+```
+
+### Effect on Triage tickets.
+To ensure that there is less noise for Triage tickets, any cluster that has events consistent with this sort of override should be omitted from Triage ticketing, in a similar way to how this is done for jupiter installations.
+
+Example here: https://github.com/openshift-assisted/assisted-installer-deployment/blob/463dc6612c923c46d47e80762c72651758a690d3/tools/create_triage_tickets.py#L64
+
+### Per customer feature enablement.
+This feature is to be enabled on a per customer basis, so will need to use the "per customer feature enablement" feature.
+
+See the following link for more information.
+
+https://github.com/openshift/assisted-service/blob/master/docs/dev/feature-per-customer.md
+
+### User Stories
+
+#### Story 1
+
+As a Kaloom user, I would like to be able to bypass VIP validation in cases where VIPS could be incorrectly shown as busy. So that I may proceed with installation.
```

Comment:
It's probably best to keep specific company names out of user stories.

Reply:
Agreed, will remove